base/inc/CopCore/include/CopCore/Global.h

```diff
@@ -27,9 +27,9 @@ enum BackendType { CPU = 0, CUDA, HIP };
 
 /** @brief CUDA error checking */
 #ifndef COPCORE_CUDA_COMPILER
-inline void error_check(int, const char *, int) {}
+static inline void error_check(int, const char *, int) {}
```

Comment:
Why not move these to a source file instead of keeping them a the header?
These are not performance critical functions, so I think moving them to a cpp/cu file is a better solution.

Reply:
I wouldn't really mind them being in a source file, but that would have to be the CUDA version, which implies that all executables need to link with the CUDA libraries and there can never be a HIP implementation, say.

Reply:
In such cases, the build system can choose a source file with the appropriate implementations depending on the backend (i.e. either a .cpp or a .cu). This is a common way to avoid having to have `#ifdefs` in a lot of places. As long as the prototype doesn't change, it should work. For now we can just use `static` if this is a pain to move to a source file. The annoying thing is that you cannot tell CMake to consider a .cpp file as a CUDA source file, only .cu files, otherwise you could just have a .cpp with both implementations too.